### ISSUE TITLE
ReactiveUI.UnhandledErrorException: An object implementing IHandleObservableErrors (often a ReactiveCommand or ObservableAsPropertyHelper) has errored, thereby breaking its observable pipeline. To prevent this, ensure the pipeline does not error, or Subscribe to the ThrownExceptions property of the object in question to handle the erroneous case.

### ISSUE THREAD

**Issue by holzerpe** [RandomEngy/VidCoder#1005]
### What were you doing when VidCoder crashed?

I pressed the button: "Encodieren". I wanted to encode a DVD.

### Crash details

```text
VidCoder 7.13 SquirrelInstaller
ReactiveUI.UnhandledErrorException: An object implementing IHandleObservableErrors (often a ReactiveCommand or ObservableAsPropertyHelper) has errored, thereby breaking its observable pipeline. To prevent this, ensure the pipeline does not error, or Subscribe to the ThrownExceptions property of the object in question to handle the erroneous case.
 ---> System.ArgumentNullException: Value cannot be null. (Parameter 'key')
   at System.Collections.Generic.Dictionary`2.FindValue(TKey key)
   at System.Collections.Generic.Dictionary`2.TryGetValue(TKey key, TValue& value)
   at VidCoder.Services.HardwareResourceService.GetRequiredHardware(EncodeJobViewModel job)
   at VidCoder.Services.HardwareResourceService.TryAcquireSlot(EncodeJobViewModel job)
   at VidCoder.Services.ProcessingService.EncodeNextJobs()
   at VidCoder.Services.ProcessingService.StartEncodeQueue()
   at VidCoder.Services.ProcessingService.<get_Encode>b__127_0()
   at ReactiveUI.ReactiveCommand.<>c__DisplayClass0_0.<Create>b__1(IObserver`1 observer) in D:\a\ReactiveUI\ReactiveUI\src\ReactiveUI\ReactiveCommand\ReactiveCommand.cs:line 109
   at System.Reactive.Linq.QueryLanguage.CreateWithDisposableObservable`1.SubscribeCore(IObserver`1 observer) in /_/Rx.NET/Source/src/System.Reactive/Linq/QueryLanguage.Creation.cs:line 35
   at System.Reactive.ObservableBase`1.Subscribe(IObserver`1 observer) in /_/Rx.NET/Source/src/System.Reactive/ObservableBase.cs:line 58
   --- End of inner exception stack trace ---
   at ReactiveUI.RxApp.<>c__DisplayClass10_0.<.cctor>b__2() in D:\a\ReactiveUI\ReactiveUI\src\ReactiveUI\RxApp.cs:line 106
   at System.Reactive.Concurrency.Scheduler.<>c.<Schedule>b__74_0(IScheduler _, Action a) in /_/Rx.NET/Source/src/System.Reactive/Concurrency/Scheduler.Simple.cs:line 37
   at System.Reactive.Concurrency.DispatcherScheduler.<>c__DisplayClass12_0`1.<Schedule>b__0() in /_/Rx.NET/Source/src/System.Reactive/Platforms/Desktop/Concurrency/DispatcherScheduler.cs:line 98
   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)
```

**Comment by RandomEngy** [comment]
Thanks for the report. For some reason the drive didn't have a root directory on it. How did you load up the DVD? Did you click on the DVD icon that shows up when you load the program? What did it look like?

I think I have a fix for the immediate crash but want to understand a bit more how that RootDirectory value became null.

**Comment by RandomEngy** [comment]
Fixed in 7.14.
